lua/lean/stderr.lua

```diff
@@ -1,17 +1,43 @@
 local log = require'vim.lsp.log'
 local stderr = {}
 
--- Show stderr output in :messages
+-- Show stderr output in separate buffer
 -- TODO: add upstream neovim API
 function stderr.enable()
   local old_error = log.error
+  local stderr_bufnr, stderr_winnr
   log.error = function(...)
     local argc = select('#', ...)
     if argc == 0 then return true end -- always enable error messages
     if argc == 4 and select(1, ...) == 'rpc' and select(3, ...) == 'stderr'
         and string.match(select(2, ...), 'lean') then
       local chunk = select(4, ...)
-      vim.schedule(function() vim.api.nvim_err_writeln(chunk) end)
+      vim.schedule(function()
+        if not stderr_bufnr or not vim.api.nvim_buf_is_valid(stderr_bufnr) then
+          stderr_bufnr = vim.api.nvim_create_buf(false, true)
+          vim.api.nvim_buf_set_name(stderr_bufnr, "lean://stderr")
+          stderr_winnr = nil
+        end
+        if not stderr_winnr or not vim.api.nvim_win_is_valid(stderr_winnr) then
+          local old_win = vim.api.nvim_get_current_win()
+          vim.cmd(('botright sbuffer %d'):format(stderr_bufnr))
```

Comment:
I'd slightly prefer always splitting this below the infoview window (i.e. not having it be full-width) which at least for the typical window layout will cause less jittering. 

Don't feel strongly though if you like it this way.

Reply:
This might be a good idea.  I'll try it out.